CI: spec update is failing

Spec update is working, but fails when it attempts to check out the latest version. For example, see https://github.com/open-telemetry/opentelemetry.io/actions/runs/5048585054/jobs/9056961958:

```diff
HEAD is now at 94196dc Blog: exponential histograms (#2661)
REPO:            opentelemetry-specification
LATEST VERSION:  1.21.0
SEARCHING for:   '^ *spec:' in scripts/content-modules/adjust-pages.pl
CURRENT VERSION:   spec: 1.20.0
+ sed -i.bak -e 's/\(^ *spec:\) .*/\1 1.21.0/' scripts/content-modules/adjust-pages.pl

Version update necessary:
diff --git a/scripts/content-modules/adjust-pages.pl b/scripts/content-modules/adjust-pages.pl
index 0be2462..b92bc[38](https://github.com/open-telemetry/opentelemetry.io/actions/runs/5048585054/jobs/9056961958#step:4:39) 100755
--- a/scripts/content-modules/adjust-pages.pl
+++ b/scripts/content-modules/adjust-pages.pl
@@ -17,7 +17,7 @@ my $semConvRef = "$otelSpecRepoUrl/blob/main/semantic_conventions/README.md";
 my $specBasePath = '/docs/specs';
 my $path_base_for_github_subdir = "content/en$specBasePath";
 my %versions = qw(
-  spec: 1.20.0
+  spec: 1.21.0
   otlp: main
 );
 my $otelSpecVers = $versions{'spec:'};

Switching to opentelemetry-specification at tag v1.21.0
+ cd content-modules/opentelemetry-specification
+ git switch --detach v1.21.0
fatal: invalid reference: v1.21.0
Error: Process completed with exit code 128.
```

I suspect that it is because the clone is shallow, and therefore doesn't have access to the tag.

--- a/.github/workflows/scripts/auto-update-version.sh
+++ b/.github/workflows/scripts/auto-update-version.sh
@@ -72,6 +72,7 @@ fi
 if [[ "$repo" == "opentelemetry-specification" ]]; then
   echo "Switching to $repo at tag v$latest_version"
   ( set -x;
+    npm run get:submodule -- content-modules/opentelemetry-specification &&
     cd content-modules/opentelemetry-specification &&
     git switch --detach v$latest_version
   )

